sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ApplicationGateway.java

```diff
@@ -638,6 +652,37 @@ interface WithWebApplicationFirewallPolicy {
             WithCreate withNewWebApplicationFirewallPolicy(Creatable<WebApplicationFirewallPolicy> creatable);
         }
 
+        /**
+         * The stage of the application gateway definition allowing to configure TLS/SSL policy for the application gateway.
+         */
+        interface WithSslPolicy {
+            /**
+             * Configures to use predefined TLS/SSL policy for the application gateway.
+             *
+             * @param policyName predefined TLS/SSL policy name
+             * @return the next stage of the definition
+             */
+            WithCreate withPredefinedSslPolicy(ApplicationGatewaySslPolicyName policyName);
+
+            /**
+             * Configures to use CustomV2 policy for the application gateway.
+             *
+             * @param minProtocolVersion minimum version of TLS/SSL protocol to be supported on application gateway.
+             * @param cipherSuites TLS/SSL cipher suites to be enabled in the specified order to application gateway.
+             * @return the next stage of the definition
+             */
+            WithCreate withCustomV2SslPolicy(ApplicationGatewaySslProtocol minProtocolVersion,
```

Comment:
Are we sure it is `V2`? Can it be used for `BASIC`?

Reply:
Good point. Let me test that out.

Reply:
From my test result, both `Custom` and `CustomV2` can be used on `Basic` or `V2` SKUs. 

Though in Portal, for `Basic` Sku, one can only configure `Custom`. For `V2` Skus however, both `Custom` and `CustomV2` can be specified.

I'll add another overload `withCustomSslPolicy`.